### PR TITLE
tracing: Fix handle leak in TracepointProvider

### PR DESCRIPTION
**Pull request by badone**
Silences Coverity 1397733

Signed-off-by: Brad Hubbard <bhubbard@redhat.com>

### CODE REVIEW

File: src/common/TracepointProvider.cc [hunk 1]
```diff
@@ -39,6 +39,7 @@ void TracepointProvider::verify_config(const struct md_config_t *conf) {
   void *handle = dlopen(m_library.c_str(), RTLD_NOW);
   if (handle != NULL) {
     m_enabled = true;
+    free(handle);
```
Comment (xiexingguo) [inline src/common/TracepointProvider.cc]: dlclose(handle)?

Comment (badone) [inline src/common/TracepointProvider.cc]: @xiexingguo that would unload the shared object and I'm not sure that's the intention here? You're right though that free is probably not the way to go in hindsight. @dillaman could we get your opinion here?

Comment (badone) [inline src/common/TracepointProvider.cc]: Perhaps making the handle a member and calling dlclose in the destructor?

**Comment by badone** [comment]
I've changed this to be more of an RAII approach which I think is better. @dillaman still like your opinion when you are around.

File: src/common/TracepointProvider.cc [hunk 2]
```diff
@@ -36,8 +39,8 @@ void TracepointProvider::verify_config(const struct md_config_t *conf) {
     return;
   }
 
-  void *handle = dlopen(m_library.c_str(), RTLD_NOW);
-  if (handle != NULL) {
+  m_handle = dlopen(m_library.c_str(), RTLD_NOW);
+  if (m_handle != NULL) {
```
Comment (dillaman) [inline src/common/TracepointProvider.cc]: Nit: I would now eliminate the ```m_enabled``` member variable and just use ```m_handle != nullptr``` to determine if it's enabled.

Comment (badone) [inline src/common/TracepointProvider.cc]: Yes, more like @dragonylffly solution in #12673 Agreed, it's a good idea.

**Comment by badone** [comment]
Updated. Many thanks to @dragonylffly for his contribution.

**Pull request merged**